shared/controller-api/src/main/java/io/pravega/shared/controller/event/AbortEvent.java

```diff
@@ -23,6 +23,7 @@
     private final String stream;
     private final int epoch;
     private final UUID txid;
+    private final long requestId;
```

Comment:
I'm puzzled as of why these `*Event` data structures are not versioned. @shiveshr ?

Reply:
All these classes have serialVersionUID added to them. 
So adding this field will be both forward and backward compatible.

Reply:
There are two problems here:

1- This is not using the general scheme that we have adopted.
2- At least for this change, we need to bump the `serialVersionUID` to indicate that there has been changes.

Reply:
I agree with problem 1. Opened issue https://github.com/pravega/pravega/issues/2994

As for `2` serialVersionUID's purpose is to tell the JVM if two different versions of the class are compatible for serialization and deserialization in backward compatible fashion. So bumping the version (changing it in any way) will tell the JVM that they are incompatible classes and it will not be able to deserialize old blobs into new classes and vice versa.

Reply:
https://howtodoinjava.com/java/serialization/serialversionuid/

```
During serialization, java runtime associates a version number with each serializable class. This number called serialVersionUID, which is used during deserialization to verify that the sender and receiver of a serialized object have loaded classes for that object that are compatible with respect to serialization.```

Reply:
Thanks to your feedback, I changed the `RequestTag.NON_EXISTENT_ID = 0L` and added checks in `SegmentHelper` to infer whether or not a new `requestId` should be generated:
```final long requestId = (clientRequestId == RequestTag.NON_EXISTENT_ID) ? idGenerator.get() : clientRequestId;```
Therefore, if an event from an "old" controller is processed by a "new" controller (with events containing `requestId`), a default value for `requestId` will be set. This default `requestId` will be logged (to notice about this situation) and used to execute operations in `SegmentHelper`, which now will generate a `requestId` exactly as it was doing before.